hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/impl/WrappedIOException.java

```diff
@@ -30,13 +31,15 @@
  * A wrapper for an IOException which
  * {@link FutureIOSupport#raiseInnerCause(ExecutionException)} knows to
  * always extract the exception.
- *
+ * <p></p>
  * The constructor signature guarantees the cause will be an IOException,
  * and as it checks for a null-argument, non-null.
+ * @deprecated: use the {@code UncheckedIOException}.
  */
+@Deprecated
 @InterfaceAudience.Private
 @InterfaceStability.Unstable
-public class WrappedIOException extends RuntimeException {
+public class WrappedIOException extends UncheckedIOException {
```

Comment:
nice touch, we were talking about this exception is a more narrow kind of runtime exception.

Reply:
I only found out about this exception recently. We can throw it in our APIs without feeling guilty. Not our fault java insists on checked exceptions